lib/pure/asyncdispatch.nim

```diff
@@ -338,7 +338,10 @@ when defined(windows) or defined(nimdoc):
       pcd: PostCallbackDataPtr
     AsyncEvent* = ptr AsyncEventImpl
 
-    Callback* = proc (fd: AsyncFD): bool {.closure, gcsafe.}
+    Callback* = proc (fd: AsyncFD): bool {.closure, gcsafe.} ## A closure execute asynchronously. 
+      ## May act upon an Async file descriptor `fd`.
+      ## Returns `true` if the callback shall be deregistered from the dispatcher.
+      ## Returns `false` if the callback shall remain registered with the dispatcher, allowing repeated execution until it is deregistered.
```

Comment:
Is this true in all cases where a `Callback` is called?

Reply:
Thanks for taking the time to look over this!

I am overall aware of 4 procs that use `Callback`:
- [addRead](https://nim-lang.org/docs/asyncdispatch.html#addRead%2CAsyncFD%2CCallback)
- [addTimer](https://nim-lang.org/docs/asyncdispatch.html#addTimer%2Cint%2Cbool%2CCallback)
- [addProcess](https://nim-lang.org/docs/asyncdispatch.html#addProcess%2Cint%2CCallback)
- [addEvent](https://nim-lang.org/docs/asyncdispatch.html#addEvent%2CAsyncEvent%2CCallback)

Of those I have only used `addTimer` so far. There I observed the behaviour that if you tell addTimer to not execute the cb only once (so oneshot = false), it'll execute again and again whenever you call `poll`:

```nim
import std/[asyncdispatch]

let echoMe: Callback = proc(fd: AsyncFD): bool = 
  echo "Me"
  return false

addTimer(100, false, echoMe)
poll() # Me
poll() # Me
poll() # Me
```
This will error out if the callback returns true. That's where I got my initial impression on what this boolean means from, which I debated further in [this nim-thread](https://forum.nim-lang.org/t/10428#69494). 

addRead and addWrite both come with these lines in their doc-comments:

> Be sure your callback cb returns true, if you want to remove watch of read notifications, and false, if you want to continue receiving notifications.

> Be sure your callback cb returns true, if you want to remove watch of write notifications, and false, if you want to continue receiving notifications.

Based on that I gained the impression that here as well the cb is put on the dispatcher and whether you return true/false determines if it gets removed from there or not. That is my current level of understanding anyway.

I am very much not sure if this holds up with `addProcess` as based on the description that's more like a trigger-proc to execute on the process' death. The source-code actually makes me confused as to why `Callback` is used as the type here at all when `proc(fd: AsyncFD): void` seems much more like it is what is desired there.

Reply:
@Varriount any feedback? Or anything I can do to get an idea on what direction is desired in regards to point 4?

Reply:
I would go ahead and remove this then. If we want to make this assurance later, we always can. The same can't be said for later removal.

Reply:
I mean, the idea is to make Callback less opaque than it is, as right now unless you basically are deep in the know, it's a mystery datatype that returns a boolean for which it is unclear what it's supposed to be for.

I want *something* there that gives me, as a user, some indication on how to use it that doesn't require me to already know how everything works.
I'm happy to change this as is, but having nothing on Callback just seems not helpful.